### PR TITLE
Add base module tests without providers and views

### PR DESCRIPTION
**Pull request by radyslavkrechet**
I've implemented `BaseViewModelSpec`, `BasePaginationViewModelSpec` and `BasePaginationViewControllerSpec` but I think I think we don't need them. We need to discuss it.

FYI: Providers and views tests will be implement in next PR

### CODE REVIEW

File: ShopApp/App/Modules/Base/Controllers/BaseTableViewController.swift [hunk 1]
```diff
@@ -26,23 +26,23 @@ class BaseTableViewController<T: BasePaginationViewModel>: BasePaginationViewCon
...
     private func setupPullToRefresh() {
         refreshControl = UIRefreshControl()
         refreshControl?.addTarget(self, action: #selector(pullToRefreshHandler), for: .valueChanged)
-        tableView.refreshControl = refreshControl
+        tableView?.refreshControl = refreshControl
     }
     
     private func setupInfinityScroll() {
-        tableView.setShouldShowInfiniteScrollHandler { [weak self] _ -> Bool in
+        tableView?.setShouldShowInfiniteScrollHandler { [weak self] _ -> Bool in
```
Comment (AntonoffEvgeniy) [inline ShopApp/App/Modules/Base/Controllers/BaseTableViewController.swift]: Can we remove `-> Bool` to use syntax like in another closures?

File: ShopAppTests/App/Base/Controllers/NavigationControllerSpec.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        }
+        
+        describe("when view loaded") {
+            it("should have correct tint colors in navigation bar") {
+                expect(navigationController.navigationBar.tintColor).to(equal(.black))
+                expect(navigationController.navigationBar.barTintColor).to(equal(.white))
+            }
+            
+            it("should have correct images in navigation bar") {
+                let shadowColor = UIColor.black.withAlphaComponent(0.12)
```
Comment (plzen) [inline ShopAppTests/App/Base/Controllers/NavigationControllerSpec.swift]: Move this to constants

File: ShopAppTests/App/Base/Controllers/TabBarControllerSpec.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        var tabBarController: TabBarController!
+        
+        beforeEach {
+            tabBarController = TabBarController()
+            _ = tabBarController.view
+        }
+        
+        describe("when view loaded") {
+            it("should have correct images in tab bar's appearance") {
+                let shadowColor = UIColor.black.withAlphaComponent(0.12)
```
Comment (plzen) [inline ShopAppTests/App/Base/Controllers/TabBarControllerSpec.swift]: Move this to constants

File: ShopAppTests/App/Base/Controllers/TabBarControllerSpec.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                let backgroundColor = UIColor(displayP3Red: 0.9765, green: 0.9765, blue: 0.9765, alpha: 0.9)
```
Comment (plzen) [inline ShopAppTests/App/Base/Controllers/TabBarControllerSpec.swift]: Move this to constants

**Pull request merged**